cmake/BLTPrivateMacros.cmake

```diff
@@ -216,7 +216,9 @@ macro(blt_setup_cuda_source_properties)
 
 
     foreach (_file ${arg_TARGET_SOURCES})
-        if (${_file} MATCHES "\\.(f|F)\\*")
+      get_source_file_property(_lang ${_file} LANGUAGE)
```

Comment:
This is the alternative location where we were trying to find Fortran files

@cyrush @white238 I think it's probably better to have this here and revert the FindCUDA changes...

Reply:
yes, that looks like a better approach!

Reply:
I think it's appropriate to exclude C in addition to Fortran here. CUDA is a C++ extension, and you'll likely get linking issues for mixed C and C++ code if you auto identify C files as CUDA files.

Reply:
Thanks @robinson96 - I'll extend the list, get find of the FindCUDA changes, then we can merge this and #82

Reply:
To avoid surprises for C cases -- we could also try to change the CMake linker lang to CXX in the case that we are using CUDA and a we have a source file that CMake initially thinks requires C linkage.

We may still have a problem if the project doesn't specify CXX as a language.

Reply:
I just ran this over a set of C++ and Fortran files for SAMRAI, and the `LANGUAGE` property was very inconsistently set. For the majority of the files scanned, it looks like it sets `_lang` to `NOTFOUND`, even for "official" fortran and C++ extensions `.f` and `.C`, respectively and with C, CXX, and Fortran languages enabled. This was using CMake 3.9.2 on an LLNL machine. I'm investigating workarounds.

Reply:
@jdahm, is SAMRAI passing LANGUAGES CXX and Fortran as part of its project commands?

Reply:
Yes, it's using `project(SAMRAI C CXX Fortran)`

Reply:
@jdahm  thats a bummer, thats the only simple issue I can think of off the top of my head

Reply:
@jdahm It sounds like a bug report to Kitware may be in order. A collaborator of mine has a support contract with them if LLNL doesn't, and we're looking at using BLT in a project together for NRC, so we may be able to justify commissioning a speedy bugfix. (Obviously it won't help for older versions of CMake though.)

@jdahm SAMRAI is up on GH, but I don't see any CMake files. If you are willing to share I'd be happy to take a look and submit a bug report to Kitware, or if you prefer, you can do it at https://gitlab.kitware.com/cmake/cmake/issues

Reply:
@zbeekman We're building in the feature/uvm branch of SAMRAI, but it requires a few fixes and dependencies, so it's not the easiest target for this. We fixed it there by manually setting the language property of the files.

I just verified and together with the workaround there this works, so I think it's OK to merge.

Reply:
@jdahm Curious what the work around was. 

Just double checked the docs for LANGUAGE property:

> A property that __*can*__ be set to indicate what programming language the source file is. If it is not set the language is determined based on the file extension.

Sounds like this is *NOT* set by CMake most of the time, but you can set it yourself. @davidbeckingsale this makes me think that an alternate solution for the diff being discussed may be required.